solana/ts/sdk/ntt.ts

```diff
@@ -349,22 +474,58 @@ export class NTT {
       shouldQueue: args.shouldQueue
     }
 
-    return await this.program.methods
+    const transferIx = await this.program.methods
       .transferBurn(transferArgs)
-      .accounts({
+      .accountsStrict({
         common: {
           payer: args.payer,
           config: { config: this.configAccountAddress() },
           mint,
           from: args.from,
+          tokenProgram: await this.tokenProgram(config),
           outboxItem: args.outboxItem,
-          outboxRateLimit: this.outboxRateLimitAccountAddress()
+          outboxRateLimit: this.outboxRateLimitAccountAddress(),
+          custody: await this.custodyAccountAddress(config),
+          systemProgram: SystemProgram.programId,
         },
         peer: this.peerAccountAddress(args.recipientChain),
         inboxRateLimit: this.inboxRateLimitAccountAddress(args.recipientChain),
-        sessionAuthority: this.sessionAuthorityAddress(args.fromAuthority, transferArgs)
+        sessionAuthority: this.sessionAuthorityAddress(args.fromAuthority, transferArgs),
+        tokenAuthority: this.tokenAuthorityAddress()
       })
       .instruction()
+
+    const mintInfo = await splToken.getMint(
+      this.program.provider.connection,
+      config.mint,
+      undefined,
+      config.tokenProgram
+    )
+    const transferHook = splToken.getTransferHook(mintInfo)
+
+    if (transferHook) {
+      const source = args.from
+      const mint = config.mint
+      const destination = await this.custodyAccountAddress(config)
+      const owner = this.sessionAuthorityAddress(args.fromAuthority, transferArgs)
+      await addExtraAccountMetasForExecute(
+        this.program.provider.connection,
+        transferIx,
+        transferHook.programId,
+        source,
+        mint,
+        destination,
+        owner,
+        // TODO(csongor): compute the amount that's passed into transfer.
+        // Leaving this 0 is fine unless the transfer hook accounts addresses
```

Comment:
It would be nice to have a test to check the error condition when token with transfer fee is used too. (the `before != after` condition in both transfer paths)